src/winxx.cc

```diff
@@ -305,12 +311,26 @@ void win_paint_tabs(HDC dc, int width) {
 
             if (active) {
                 auto _f = SelectWObj(bufdc, new_active_tab_font());
-                paint_tab(bufdc, tabwidth, tabs[i]);
+                paint_tab(bufdc, tabwidth, loc_tabheight, tabs[i]);
             } else {
-                paint_tab(bufdc, tabwidth, tabs[i]);
+                MoveToEx(bufdc, 0, loc_tabheight, nullptr);
+                LineTo(bufdc, tabwidth, loc_tabheight);
+                paint_tab(bufdc, tabwidth, loc_tabheight, tabs[i]);
             }
 
-            BitBlt(dc, i*tabwidth, 0, tabwidth, tabheight+1,
+            BitBlt(dc, i*tabwidth+1, 1, tabwidth, tabheight(),
```

Comment:
I think it's possible here that the source buffer (`bufdc`) goes over the right edge of target buffer (`dc`) because of the off-by-one error here (`i*tabwidth+1`). For n tabs with the width that equals to `width/n` the widtht of all tabs combined will be `width+1` which is  too much.

This might not cause segmentation fault though, since at least msdn claims BitBlt performs clipping for destination DC. But I think this is still slightly wrong.